### ISSUE TITLE
Improve search interface

### ISSUE THREAD

**Issue by tobiasdiez** [JabRef/jabref#7423]
The search ui interface needs some attention. The current author-auto completion lives between broken and useless. 
My proposal would be to add a proper flyout that can now be seen also in some other applications (e.g. OneNote or Outlook).
I particularly like the following:

![image](https://images.ctfassets.net/dkgr2j75jrom/4luapfPtOuKsGM91VAPd9B/55f9d8e33a0782df71f9bf602aeb49e0/fs_Omnisearch.svg)
from https://www.fullstory.com/platform, see also https://help.fullstory.com/hc/en-us/articles/360020828253-Getting-Acquainted-with-OmniSearch

**Comment by koppor** [comment]
The other proposal by @tobiasdiez (at https://github.com/JabRef/jabref/issues/3535#issuecomment-352068826) is written down there: https://github.com/koppor/jabref/issues/341

**Comment by teertinker** [comment]
Hello, if there will be any change of the search function please keep the missing "global search" function in mind. From my point of view it is still a pain to enter the search term for every single tab/bib I want to search.

**Comment by koppor** [comment]
The global search is issue https://github.com/JabRef/jabref/issues/4096 (and not directly related to this issue - and thus can be handled separatetly).

**Comment by mchellelm** [comment]
Hi, we're a team of 5 (@shameenkals @JasonBourne999 @lachyp @artherdunger10) from the University of Adelaide who would like to tackle this issue! Our goal will be improving the author-auto completion as well as implementing a proper fly-out as proposed in this issue. We aim to have a pull request by early May. Is there anything in particular we should know about?

**Comment by tobiasdiez** [comment]
Thanks for your interest! 

We are currently in the process of migrating the search syntax to Lucene (see #8206 for ongoing work). So in early prototypes its a good idea to not fiddle to much with the search syntax parsing. Please also have a look at the other linked issues (in particular https://github.com/koppor/jabref/issues/341) for further inspiration.

Our experience shows that its usually a good idea to create a PR very early during the development of a feature. You can then use it as a place to ask questions that might come up, and we can give feedback about the general direction the PR is heading to. Please feel free to contact us through one of the various channels if questions arise.

**Comment by a1746905** [comment]
Hi, I'm one of the members from the team of students in @mchellelm's group attempting to tackle this issue. So far we've been reading the documentation and experimenting with the code to try and figure out what changes specifically need to be made to create a search interface like the one pictured at the start of this issue and we just have a few questions:

1. How would you suggest we start with conducting searches with multiple different categories on the AutoCompleter?"
2. Given that the results of an auto-completion are dynamically stored in an ObservableList object, how would you suggest we go about adding headings to this list when its rendered in the GUI? Would this be done via java or via CSS?

**Comment by LachyP** [comment]
Hi, I'm from the team of students in @mchellelm's group attempting to tackle this issue. So far we've been reading the documentation and experimenting with the code to try and figure out what changes need to be made to create a search interface like the one pictured at the start of this issue and we just have a few questions:
1. How would you suggest we start with conducting searches with multiple different categories on the AutoCompleter?
2. Given that the results of an auto-completion are dynamically stored in an ObservableList object, how would you suggest we go about adding headings to this list when its rendered in the GUI? Would this be done via java or via CSS?

**Comment by tobiasdiez** [comment]
I would say the headings should be static and predefined (at least for now). A good starting point would be to have a single category "Authors" whose contents is a filtered list of author names as provided by the auto completer. When the user selects one of the options, then something like `authors = xyz` should be added to the search text. Then as a follow-up further headings (such as "Priority") can be added with values either hard coded (prio1, prio2, ...) or from some auto completer (which then probably has to be newly implemented). 

In general, one defines the structure of the control in java (or fxml, which I wouldn't suggest for this component here due to the performance overhead) and how it looks in css.

Does this answers your questions?

**Comment by LachyP** [comment]
Yes that makes sense, thankyou. I'll let you know if we have any more troubles

**Comment by koppor** [comment]
Pre-condition: https://github.com/JabRef/jabref/pull/8963

**Comment by priyanshu16095** [comment]
Can I work on this issue?

**Comment by priyanshu16095** [comment]
Please assign me this issue.
/assign-me

**Comment by koppor** [comment]
Some other interface idea

<img width="1280" height="901" alt="Image" src="https://github.com/user-attachments/assets/216d3f3e-a870-4e44-90b0-a4ff3c014d4e" />

**Comment by koppor** [comment]
Zotero - https://github.com/milekpl/zotero-search-replace

<img width="1292" height="1061" alt="Image" src="https://github.com/user-attachments/assets/409649c6-76cf-4236-ba2b-6223097916a6" />

**Comment by priyanshu16095** [comment]
Please someone assign me this issue, I am gathering ideas and want to work on this.

**Comment by koppor** [comment]
<img width="661" height="351" alt="Image" src="https://github.com/user-attachments/assets/fc3114b2-7d87-4934-a554-4c36003e2afb" />

https://sourceforge.net/projects/biblioassist/

Found by @Siedlerchr
